var/spack/repos/builtin/packages/treelite/package.py

```diff
@@ -3,14 +3,58 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+import os
 from spack import *
 
 
-class Treelite(CMakePackage):
+class Treelite(CMakePackage, PythonPackage):
     """Treelite is a model compiler for efficient deployment of
     decision tree ensembles."""
 
     homepage = "https://github.com/dmlc/treelite"
     url      = "https://github.com/dmlc/treelite/archive/0.93.tar.gz"
 
     version('0.93',    sha256='7d347372f7fdc069904afe93e69ed0bf696ba42d271fe2f8bf6835d2ab2f45d5')
+
+    variant('protobuf', default=False, description='Build with protobuf')
+    variant('python', default=True, description='Build with python support')
+
+    depends_on('protobuf', when='+protobuf')
+    depends_on('python@3.6:', when='+python', type=('build', 'run'))
+    depends_on('py-setuptools', when='+python', type='build')
+    depends_on('py-numpy', when='+python', type=('build', 'run'))
+    depends_on('py-scipy', when='+python', type=('build', 'run'))
+
+    build_directory = 'build'
+    phases = ['cmake', 'build', 'install', 'python_build', 'python_install']
+
+    def cmake_args(self):
+        args = []
+
+        if '+protobuf' in self.spec:
+            args.append('-DENABLE_PROTOBUF:BOOL=ON')
+            args.append('-DProtobuf_LIBRARY={0}'.format(
+                self.spec['protobuf'].prefix))
+        else:
+            args.append('-DENABLE_PROTOBUF:BOOL=OFF')
+
+        return args
+
+    def python_build(self, spec, prefix):
+        if '+python' in spec:
+            self._build_directory = 'python'
+            PythonPackage.build_ext(self, spec, prefix)
+        else:
+            print('python deselected')
+
+    def python_install(self, spec, prefix):
+        if '+python' in spec:
+            PythonPackage.install(self, spec, prefix)
+        else:
+            print('python deselected')
```

Comment:
Instead of adding new phases I would maybe keep the default CMake phases and use `@run_after('build')` and `@run_after('install')`

Reply:
Is there a difference between the two ways of doing it, that would cause a functional change? I do see them as separate phases and having the build log show that is helpful.

I could swap `python_build` and `install` in the phase order. That might make more sense than how I have it written.

Reply:
No functional difference except that `spack install --until` would have fewer phases it could stop at.